src/Elmish.WPF/ViewModel.fs

```diff
@@ -107,37 +107,38 @@ type internal TwoWayBinding<'model> = {
   Set: obj -> 'model -> unit
 }
 
-type internal SubModelBinding<'model, 'msg, 'bindingModel, 'bindingMsg> = {
-  SubModelData: SubModelData<'model, 'msg, 'bindingModel, 'bindingMsg>
-  Vm: ViewModel<'bindingModel, 'bindingMsg> voption ref
+type internal SubModelBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel> = {
+  SubModelData: SubModelData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>
+  Vm: 'bindingViewModel voption ref
 }
 
-and internal SubModelWinBinding<'model, 'msg, 'bindingModel, 'bindingMsg> = {
-  SubModelWinData: SubModelWinData<'model, 'msg, 'bindingModel, 'bindingMsg>
+and internal SubModelWinBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel> = {
+  SubModelWinData: SubModelWinData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>
   WinRef: WeakReference<Window>
   PreventClose: bool ref
-  VmWinState: WindowState<ViewModel<'bindingModel, 'bindingMsg>> ref
+  VmWinState: WindowState<'bindingViewModel> ref
 }
 
-and internal SubModelSeqUnkeyedBinding<'model, 'msg, 'bindingModel, 'bindingMsg> = {
-  SubModelSeqUnkeyedData: SubModelSeqUnkeyedData<'model, 'msg, 'bindingModel, 'bindingMsg>
-  Vms: ObservableCollection<ViewModel<'bindingModel, 'bindingMsg>>
+and internal SubModelSeqUnkeyedBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel> = {
+  SubModelSeqUnkeyedData: SubModelSeqUnkeyedData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel>
+  Vms: ObservableCollection<'bindingViewModel>
 }
 
-and internal SubModelSeqKeyedBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'id when 'id : equality> =
-  { SubModelSeqKeyedData: SubModelSeqKeyedData<'model, 'msg, 'bindingModel, 'bindingMsg, 'id>
-    Vms: ObservableCollection<ViewModel<'bindingModel, 'bindingMsg>> }
+and internal SubModelSeqKeyedBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel, 'id when 'id : equality> =
+  { SubModelSeqKeyedData: SubModelSeqKeyedData<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel, 'id>
+    Vms: ObservableCollection<'bindingViewModel> }
 
   member d.FromId(id: 'id) =
     d.Vms
-    |> Seq.tryFind (fun vm -> vm.CurrentModel |> d.SubModelSeqKeyedData.GetId |> (=) id)
+    |> Seq.tryFind (fun vm -> vm |> d.SubModelSeqKeyedData.GetUnderlyingModel |> d.SubModelSeqKeyedData.GetId |> (=) id)
 
-and internal SubModelSelectedItemBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'id> =
+and internal SubModelSelectedItemBinding<'model, 'msg, 'bindingModel, 'bindingMsg, 'bindingViewModel, 'id> =
   { Get: 'model -> 'id voption
     Set: 'id voption -> 'model -> unit
     SubModelSeqBindingName: string
     GetId: 'bindingModel -> 'id
-    FromId: 'id -> ViewModel<'bindingModel, 'bindingMsg> option }
+    FromId: 'id -> 'bindingViewModel option
+    GetUnderlyingModel: 'bindingViewModel -> 'bindingModel }
```

Comment:
How about `GetBindingModel`?

Reply:
That name would seem to indicate that the signature is `'model -> 'bindingModel` as in you are getting a sub model. I used the term "Underlying" to differentiate and indicate that you are getting the model associated with a given `'viewModel`.

Reply:
Ok